sdk/keyvault/keyvault-certificates/test/internal/lroUnexpectedErrors.spec.ts

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as assert from "assert";
+import { DeleteCertificatePoller } from "../../src/lro/delete/poller";
+import { RecoverDeletedCertificatePoller } from "../../src/lro/recover/poller";
+
+describe("The LROs properly throw on unexpected errors", () => {
+  const vaultUrl = `https://keyVaultName.vault.azure.net`;
+
+  describe("delete LRO", () => {
+    it("403 doesn't throw", async function() {
+      const code = 403;
+      const client: any = {
+        async deleteCertificate(): Promise<any> {
+          return {
+            id: "/version/name/version",
+            recoveryId: "something"
+          };
+        },
+        async getDeletedCertificate(): Promise<any> {
+          const error = new Error(`${code}`);
```

Comment:
nit: if this was a `RestError` we could set the statusCode without casting to any right?